File: backend/src/AMS.Domain/Entities/Submission.cs
using AMS.Domain.Common;
using AMS.Domain.Enums;

namespace AMS.Domain.Entities;

public class Submission : BaseEntity
{
    public Guid AssignmentId { get; set; }

    public Guid StudentId { get; set; }

    public string Answer { get; set; } = string.Empty;

    public DateTime SubmittedAt { get; set; }

    public decimal? Marks { get; set; }

    public string? Feedback { get; set; }

    public SubmissionStatus Status { get; set; }
    
}

File: frontend/src/features/teacher/assignments/types/submission.ts
export type SubmissionStatus = "Submitted" | "Reviewed";

export interface AssignmentSubmission {
  id: string;
  assignmentId: string;
  studentId: string;
  answer: string;
  fileUrl: string | null;
  submittedAt: string;
  marks: number | null;
  feedback: string | null;
  status: SubmissionStatus;
  reviewedAt: string | null;
}
